homeassistant/components/local_ip/sensor.py

```diff
@@ -1,33 +1,32 @@
 """Sensor platform for local_ip."""
 
-from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity import Entity
 from homeassistant.util import get_local_ip
 
+from .const import DOMAIN, SENSOR
 
-async def async_setup_entry(hass: HomeAssistant, config_entry, async_add_entities):
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the platform from config_entry."""
-    name = config_entry.data["name"]
```

Comment:
We should make this backwards compatible and look for name in the config_entry.data and default to the domain. At least for one release cycle so that existing installations have the entity registry activated for this integration. After that we could change name without breaking change.